Documentation/Books/AQL/Functions/Document.md

```diff
@@ -47,6 +47,34 @@ FOR attributeArray IN attributesPerDocument
         RETURN {attr, count}
 ```
 
+CHECK_DOCUMENT()
+----------------
+
+`CHECK_DOCUMENT(document) → checkResult`
+
+Returns true if *document* is a valid document object, i.e. a document without any
+duplicate attribute names. Will return *false* for any non-objects/non-documents,
```

Comment:
This check is carried out recursively right? This should be mentioned.

Reply:
Our terminology is probably not consistent in the documentation, but I would call the top-level object _document_ and nested objects _objects_. Following this convention, CHECK_**DOCUMENT**() could be misunderstood as a check of the top-level object.

We have some operations in AQL which only work on the top-level, like `UPDATE doc WITH { ... } IN coll`. Some users tried stuff like `UPDATE subattr WITH { ... } IN coll.doc.attr`, which is not supported. AQL functions like ATTRIBUTES() don't recurse into sub-attributes, and you may have a mutable attribute called `_key` as long as it's not on the top-level of a document. Thus, I am not sure if it's clear that CHECK_DOCUMENT() is recursive.

Reply:
Maybe it's not, but REMOVE also removes the entire document and not just the top-level of it...
We are really nit-picking here, and these PRs have already taken far longer than they are worth.
Remember this is a debug functionality that is supposed to be used mainly by us, in a relatively seldom case.